packages/server-runtime/src/index.ts

```diff
@@ -0,0 +1,24 @@
+import { type AnyFunction, type SolenoidFunctionConfig, SOLENOID_CUSTOM_KEY, SOLENOID_OBJECT_TYPES } from "@solenoid/custom-elements/dist/utils/types";
+
+type SerializableFn<C extends any[] = any[], A extends AnyFunction = AnyFunction> = (...c: C)=>A;
+
+type SerializableFnConf<F extends SerializableFn> = {
+  fn: F,
+
+  closure: ()=>Parameters<F>,
+  id: string,
+};
+
+export function serializableFn<T extends SerializableFn>({fn, closure, id}: SerializableFnConf<T>): ReturnType<T> & SolenoidFunctionConfig<Parameters<T>> {
+  const closureVars = closure();
+  const fnWithClosuresFilled = fn(...closureVars) as ReturnType<typeof fn>;
```

Comment:
A lot of these types are effectively useless because this is a compile target function and will never be used with types.